subsys/debug/etb_trace/etb_trace_lp.c

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright (c) 2023 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <zephyr/kernel.h>
+#include <zephyr/wait_q.h>
+#include <debug/etb_trace.h>
+
+#include <etb_trace_private.h>
+
+/* Flag to signal that we suspect a long enough sleep duration to justify turning off tracing. */
+static bool trace_stopped;
+
+/**
+ * @brief Resume tracing on wakeup if it was stopped previously.
+ *
+ */
+void sys_clock_idle_exit(void)
+{
+	if (trace_stopped) {
+		etb_trace_start();
+
+		trace_stopped = false;
+	}
+}
+
+/**
+ * @brief Turn off debug unit and tracing if going to sleep for a long period of time.
+ *
+ * @return always true to allow going to idle
+ */
+bool z_arm_on_enter_cpu_idle(void)
+{
+	/* Stop only if traces are enabled */
```

Comment:
The comment and the code do not align well now. Perhaps the `ETB_FORMATTER_STOPPED` should be called `ETB_TRACE_DISABLED`?

Or may be check the state of `trace_stopped`?